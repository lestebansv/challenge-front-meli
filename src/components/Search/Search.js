import React, { useState } from 'react';
import { useNavigate } from 'react-router-dom';
import iconSearch from '../../assets/image/ic_Search.png';
import logo from '../../assets/image/Logo_ML.png';
import styles from './Search.module.scss';

/**
 * Component builds the box search
 * 
 * @returns Render view box product search  
 */
const Search = () => {

    const [textValue, setTextValue] = useState('');
    let navigate = useNavigate();

    const handlerKeyPress = (event) => {
        if (event.key === 'Enter') handlerSubmit(event);
    }

    const handlerSubmit = (event) => {
        event.preventDefault();
        if (textValue) navigate(`/items/?search=${textValue}`);
    }

    return (
        <div className={styles.header_navbar}>
            <div className={styles.header_search}>

                <div>
                    <a href="/">
                        <img className={styles.logo} src={logo} alt="Mercado Libre" />
                    </a>
                </div>

                <div>
                    <input
                        type="text"
                        className={styles.text}
                        placeholder="Nunca dejes de buscar"
                        value={textValue}
                        onChange={(event) => setTextValue(event.target.value)}
                        onKeyPress={(event) => handlerKeyPress(event)}
                    >
                    </input>
                </div>

                <div>
                    <button
                        className={styles.button_search}
                        onClick={handlerSubmit}
                    >
                        <img src={iconSearch} alt="Mercado Libre" />
                    </button>
                </div>
            </div>
        </div>
    )
}

export default Search;