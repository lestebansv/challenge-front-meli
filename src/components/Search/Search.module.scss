@import '../../assets/css/styles.scss';

.button_search {
    appearance: none;
    background: #eeeeee;
    border: 0;
    border-radius: 0 5px 5px 0;
    color: #fff;
    cursor: pointer;
    height: 36px;
    margin: 0;
    padding: 9px;
}

.logo {
    margin-right: 25px;
    vertical-align: middle;
}

.header_navbar {
    background-color: $background-yellow;
    padding: 0.5rem 0;
    width: 100%;
}

.header_search {
    display: grid;
    grid-template-columns: 55px 1fr 35px;
    margin: 0 15%;
}

.submit {
    background: $background-gray;
    border: 0;
    color: $background-white;
    height: 25px;
    margin-left: -35px;
    width: 35px;
    -webkit-appearance: none;
}

.text {
    border: 1px solid $background-gray-light;
    border-radius: 3px;
    margin-left: 15px;
    padding: 10px;
    width: -webkit-fill-available;
}

.text:focus-visible {
    outline: none;
}

.text::placeholder {
    color: $background-gray;
}