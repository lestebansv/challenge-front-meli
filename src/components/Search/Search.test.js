import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { configure, shallow } from 'enzyme';
import { render } from '@testing-library/react';
import Search from './Search';

configure({ adapter: new Adapter() });

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: jest.fn(() => 'bar')
}));

describe('Search', () => {
    const view = render(<Search />);

    it('render', () => {
        expect(view.container).toMatchSnapshot();
    });

    it('should keypress Enter input text', () => {
        const onKeyPressMock = jest.fn();

        const event = {
            key: 'Enter',
            preventDefault() { },
            target: { value: 'kindle' }
        };

        const component = shallow(<Search onKeyPress={onKeyPressMock} />);

        component.find('input').simulate('keyPress', event);
    });

    it('should keypress Enter input text value', () => {
        const onKeyChangeMock = jest.fn();

        const event = {
            key: 'Enter',
            preventDefault() { },
            target: { value: 'kindle' }
        };

        const component = shallow(<Search onChange={onKeyChangeMock} />);

        component.find('input').simulate('change', event);
    });

    it('should keypress "a" text', () => {
        const onKeyPressMock = jest.fn();

        const event = {
            key: 'A',
            preventDefault() { },
            target: { value: 'kindle' }
        };

        const component = shallow(<Search onKeyPress={onKeyPressMock} />);

        component.find('input').simulate('keyPress', event);
    });
});