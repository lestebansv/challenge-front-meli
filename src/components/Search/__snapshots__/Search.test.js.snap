// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search render 1`] = `
<div>
  <div
    class="header_navbar"
  >
    <div
      class="header_search"
    >
      <div>
        <a
          href="/"
        >
          <img
            alt="Mercado Libre"
            class="logo"
            src="Logo_ML.png"
          />
        </a>
      </div>
      <div>
        <input
          class="text"
          placeholder="Nunca dejes de buscar"
          type="text"
          value=""
        />
      </div>
      <div>
        <button
          class="button_search"
        >
          <img
            alt="Mercado Libre"
            src="ic_Search.png"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
